src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -403,6 +403,23 @@ public virtual bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType
             return CheckTypesCore(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
         }
 
+        /// <summary>
+        /// Perform sub-expression type checking and produce a return type for the function declaration, this is only applicable for UDFs.
+        /// </summary>
+        public virtual bool CheckTypesOnDeclaration(CheckTypesContext context, TexlNode[] args, DType[] argTypes, DType returnType, IErrorContainer errors, out DType actualReturnType)
+        {
+            actualReturnType = DType.Unknown;
```

Comment:
Why would we need to set the actual return type here? Isn't the body return type calculated elsewhere?

Reply:
Yeah right, since its an out parm, we either assign null or Unknown

Reply:
Why does this out param even exist? What is it doing?

Reply:
Once the body is bound, we will know the actual return type and we are returning it if the call wants to do anything with it.